File: CPP/Peripheral.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

// Kelas Peripheral yang mewarisi dari kelas Komponen
class Peripheral : public Komponen {
    private:
        string tipePeripheral;
    
    public:
        Peripheral() {}

        // Konstruktor dengan parameter untuk menginisialisasi tipePeripheral, merk, dan nama
        Peripheral(string tipePeripheral, string merk, string nama) 
            : Komponen(merk, nama) {
            this->tipePeripheral = tipePeripheral;
        }
    
        // Setter
        void setTipePeripheral(string tipePeripheral) {
            this->tipePeripheral = tipePeripheral;
        }
        
        // Getter
        string getTipePeripheral() {
            return tipePeripheral;
        }
    
    ~Peripheral()
    {
        
    }
};

File: python/Peripheral.py
from Komponen import Komponen

class Peripheral(Komponen):
    #atribut private
    __tipe_peripheral = ""

    #constructor
    def __init__(self, tipe_peripheral, merk, nama):
        super().__init__(merk, nama)
        self._tipe_peripheral = tipe_peripheral

    #setter
    def set_tipe_peripheral(self, tipe_peripheral):
        self._tipe_peripheral = tipe_peripheral

    #getter
    def get_tipe_peripheral(self):
        return self._tipe_peripheral